ocaml/testsuite/tests/parsetree/source.ml

```diff
@@ -7448,34 +7448,3 @@ let g y =
   f ~(y:int)
 
 let goober a = match a with C (type a b) y -> y
-
-(* Pattern guards *)
-let f g x = function
-  | A x when g x match B y -> y
-  | _ -> x
-
-let f g h j x = function
-  | A x
-      when g x match B y
-      when h y match C z
-      when j z match D a ->
-      a
-  | _ -> x
-;;
-
-let f g = function
-  | A x when g x match (
-    | A y when g y match (
-      | A z -> z
-      | B z -> z
-      )
-    | B y when g y match (
-      | A z -> z
-      | B z -> z
-      )
-    )
-  | B x when g x match (
-    | A y when g y match A z -> z
-    | B y when g y match A z -> z
-    )
-  | C x -> x
```

Comment:
Are these tests interesting in any way? Should we try to preserve them in the intermediate world where this feature is not yet sent upstream?

Reply:
As I understand, the purpose of this file is to test the parser and source printer. I'm not sure if these tests are particularly interesting, but we had a reason for adding them in #1588: do you remember what that was?

Reply:
I now remember: these tests check that parsing and source dumping roundtrips. This remains important, right?

Reply:
Good call, done!

(Upon initially reading this comment, I didn't realize that the PR deletes these tests)